olive/evaluator/olive_evaluator.py

```diff
@@ -146,6 +151,28 @@ def evaluate(
                 raise TypeError(f"{metric.type} is not a supported metric type")
         return flatten_metric_result(metrics_res)
 
+    @staticmethod
+    def fill_metric_user_config_with_model_io(metric: Metric, model: OliveModel):
+        # if the io_config is not specified in the metrics, use the one in the model
+        if metric.data_config:
+            return metric
+
+        io_config = model.get_io_config()
+        if not io_config:
+            return metric
+
+        if not is_io_config_static(io_config):
+            logger.debug(
+                "Model input shapes are not static. Cannot use inferred input shapes for creating dummy data. This will"
+                " cause an error when creating dummy data for tuning."
+            )
+        if io_config and not metric.user_config.input_names and not metric.user_config.input_shapes:
+            # input_types is optional
+            metric.user_config.input_names = io_config["input_names"]
+            metric.user_config.input_shapes = io_config["input_shapes"]
+            metric.user_config.input_types = io_config.get("input_types")
```

Comment:
As input_types is optional, default value is float32, we use `get` here to handle the case of `KeyNotExists`

Reply:
if the input_types doesn't exist, should we ignore the assignment or assign None?

Reply:
The logic is like:
1. if user provide 'input_types' we use that.
2. if not, use `float32` as default.

Reply:
I suppose io_config.get("input_types") would return None if no input_types?

Reply:
Yes, see this under `dataset.py`
![image](https://github.com/microsoft/Olive/assets/13343117/ef98bf6d-cff9-4096-be05-dda21b3fca2f)

Reply:
better to add comments here for the None assignment.